src/opus.py

```diff
@@ -304,9 +377,22 @@ def validate_references(opus: Opus, exit_on_failure: bool):
 
     # Actions
     referred_actions = set()
+    def update_referred_action(action: ActionTemplate):
```

Comment:
Same

Reply:
This one accesses multiple variables from the parent function. Breaking it out would require to pass them as arguments. I actually think I like it as an inline function here?

Reply:
Fair enough